### PR TITLE
Horizontal support

### LINKED ISSUE #1: Support horizontal dragging within a single list

### PR DESCRIPTION
**Pull request by alexreardon**
This pull request will add horizontal dragging support to single lists.

This will close #1 and #5 

There are still a few things to do:

- [x] fix broken tests
- [x] add tests for new behaviour and functionality
- [x] cleanup code and ensure linting / typing is correct
- [x] squash commits
- [x] update docs

### CODE REVIEW

**Comment by alexreardon** [comment]
This will be a breaking change due to the changes to the `hooks.onDragStart` argument change. I will add the reasoning for this to the reason notes

File: src/state/action-creators.js [hunk 1]
```diff
@@ -269,15 +269,25 @@ export const drop = () =>
...
 
     const result: DropResult = {
       draggableId: current.id,
+      type: current.type,
       source: initial.source,
       destination: impact.destination,
     };
 
-    const scrollDiff = getScrollDiff(initial, current, droppable);
+    // TODO: check
```
Comment (alexreardon) [inline src/state/action-creators.js]: remove TODO

File: src/state/action-creators.js [hunk 2]
```diff
@@ -286,6 +296,7 @@ export const drop = () =>
       droppableScrollDiff: scrollDiff.droppable,
       windowScrollDiff: scrollDiff.window,
       draggables: state.dimension.draggable,
+      axis: destinationDroppable ? destinationDroppable.axis : null,
```
Comment (alexreardon) [inline src/state/action-creators.js]: add test to get-new-home-client-offset to ensure that it returns the default when null is passed in

File: src/state/dimension.js [hunk 3]
```diff
@@ -1,10 +1,12 @@
 // @flow
+import { vertical, horizontal } from './axis';
```
Comment (alexreardon) [inline src/state/dimension.js]: add test to ensure that axis is applied correctly

File: src/types.js [hunk 4]
```diff
@@ -13,6 +13,16 @@ export type Position = {|
   y: number,
 |};
 
+export type Direction = 'horizontal' | 'vertical';
+
+export type Axis = {|
```
Comment (alexreardon) [inline src/types.js]: split into VerticalAxis and HorizontalAxis

type Axis = VerticalAxis | HorizontalAxis

no need to expose VerticalAxis or HorizontalAxis.

This prevents the case where an axis is made that has a mixture of vertical and horizontal properties
